configure

```diff
@@ -133,13 +141,19 @@ if [ -z "$CC" ] ; then
 	CC=cc
 fi
 
-check_compile 'whether netinet/in.h defines s6_addr16' "" \
-'#include <netinet/in.h>\nint main(int a, char**c){struct in6_addr x={.s6_addr32[0]=a};return x.s6_addr16[0]; }' \
-|| {
-check_compile 'whether netinet/in.h defines __u6_addr.__u6_addr16' \
-'-Ds6_addr16=__u6_addr.__u6_addr16 -Ds6_addr32=__u6_addr.__u6_addr32' \
-'#include <netinet/in.h>\nint main(int a, char**c){struct in6_addr x={.s6_addr32[0]=a};return x.s6_addr16[0]; }'
-}
+for i in 16 32; do
```

Comment:
why check both ? if one of the two exist, we can assume the other does too

Reply:
Check for both is needed since Solaris does only define `_S6_un._S6_u32`, not `_S6_un._S6_u16`

Reply:
can you show the snippet from the solaris header, or even better, a link to the entire header ?

Reply:
https://github.com/nxmirrors/onnv/blob/master/usr/src/uts/common/netinet/in.h
This OpenSolaris header may different from Solaris 11.3, but the `struct in6_addr` definition is same.

Reply:
thanks. it seems it was not a good idea to use s6_addr16/32, so i pushed 2 changes. can you rebase your changes on top of that tell me if it fixes this particular issue ?